### PR TITLE
Add Favourite Place Category

### PR DESCRIPTION
**Pull request by rossey**

### CODE REVIEW

File: src/Transport/Entities/FavouritePlace.php [hunk 1]
```diff
@@ -13,6 +13,7 @@ class FavouritePlace implements LabeledWaypointInterface
     protected $geolocation;
     protected $stops = [];
     protected $permanent = false;
+    protected $icon;
```
Comment (dave1010) [inline src/Transport/Entities/FavouritePlace.php]: Icon is an implementation detail of the client UI. The domain only knows that there's a category (which may include an enum and a label).

File: src/Transport/Entities/FavouritePlaceIcon.php [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    /** @return string */
+    public function getCategory()
+    {
+        return $this->category;
+    }
+
+    /** @param string $category */
+    public function setCategory($category)
+    {
+        $this->category = strtolower($category);
```
Comment (dave1010) [inline src/Transport/Entities/FavouritePlaceIcon.php]: We're saying $category has to be a certain format. Why is that? Should we then remove  `[^\w]` too?

File: tests/Transport/Normalizers/FavouritePlaceIconNormalizerTest.php [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    public function testNormalization()
+    {
+        $icon = new FavouritePlaceIcon;
+        $icon->setLabel('Label');
+        $icon->setCategory('CATEGORY');
+
+        $normalized = $this->serializer->normalize($icon);
+        $this->assertCount(2, $normalized);
+        $this->assertEquals('Label', $normalized['label']);
+        $this->assertEquals('category', $normalized['category']);
```
Comment (dave1010) [inline tests/Transport/Normalizers/FavouritePlaceIconNormalizerTest.php]: This is testing `FavouritePlaceIcon`'s setter. If that needs testing then it should be in a different test.

File: src/Transport/Entities/FavouritePlace.php [hunk 4]
```diff
@@ -81,4 +82,20 @@ public function isPermanent()
...
+    /**
+     * @param FavouritePlaceCategory $category
+     */
+    public function setCategory(FavouritePlaceCategory $category)
+    {
+        $this->category = $category;
+    }
+
+    /**
+     * @return FavouritePlaceCategory
```
Comment (dave1010) [inline src/Transport/Entities/FavouritePlace.php]: I can make this return `null` too

**Comment by dave1010** [comment]
:+1:

**Comment by rossey** [comment]
@acleon could you take a look and review this please?

**Comment by acleon** [comment]
Can we just clean up the "$icon" variables. There's several $icon variables which contain a FavouritePlaceCategory object.

**Comment by rossey** [comment]
Good spot. The only thing I can't rename is the branch :)

**Pull request merged**